spec/input_objects/pages/long_lists_selection/bulk_options_input_spec.rb

```diff
@@ -0,0 +1,102 @@
+  let(:draft_question) { build :draft_question, answer_type: "selection" }
+
+  it "has a valid factory" do
+    expect(bulk_options_input).to be_valid
+  end
+
+  describe "validations" do
+    it "is invalid if fewer than 2 Bulk selection options are provided" do
+      bulk_options_input.bulk_selection_options = "A single option"
+      error_message = I18n.t("activemodel.errors.models.pages/long_lists_selection/bulk_options_input.attributes.bulk_selection_options.minimum")
+      expect(bulk_options_input).not_to be_valid
+
+      expect(bulk_options_input.errors.full_messages_for(:bulk_selection_options)).to include("Bulk selection options #{error_message}")
+    end
+
+    it "is invalid if more than 1000 Bulk selection options are provided" do
+      bulk_options_input.bulk_selection_options = (1..1001).to_a.join("\n")
+      error_message = I18n.t("activemodel.errors.models.pages/long_lists_selection/bulk_options_input.attributes.bulk_selection_options.maximum")
+      expect(bulk_options_input).not_to be_valid
+
+      expect(bulk_options_input.errors.full_messages_for(:bulk_selection_options)).to include("Bulk selection options #{error_message}")
+    end
+
+    it "is valid if there are between 2 and 20 unique selection values" do
+      bulk_options_input.bulk_selection_options = (1..2).to_a.join("\n")
+
+      expect(bulk_options_input).to be_valid
+      expect(bulk_options_input.errors.full_messages_for(:bulk_selection_options)).to be_empty
+
+      bulk_options_input.bulk_selection_options = (1..1000).to_a.join("\n")
+
+      expect(bulk_options_input).to be_valid
+      expect(bulk_options_input.errors.full_messages_for(:bulk_selection_options)).to be_empty
+    end
+
+    context "when not given a draft_question" do
+      let(:draft_question) { nil }
+
+      it "is invalid" do
+        expect(bulk_options_input).to be_invalid
+      end
+    end
+
+    context "when include_none_of_the_above is not 'true' or 'false'" do
+      let(:bulk_options_input) { build :bulk_options_input, include_none_of_the_above: nil }
+
+      it "is invalid" do
+        error_message = I18n.t("activemodel.errors.models.pages/long_lists_selection/bulk_options_input.attributes.include_none_of_the_above.inclusion")
+        expect(bulk_options_input).to be_invalid
+        expect(bulk_options_input.errors.full_messages_for(:include_none_of_the_above)).to include("Include none of the above #{error_message}")
+      end
+    end
+  end
+
+  describe "#submit" do
+    it "returns false if the form is invalid" do
+      bulk_options_input.bulk_selection_options = ""
+      expect(bulk_options_input.submit).to be_falsey
+    end
+
+    it "sets draft_question answer_settings and is_optional" do
+      bulk_options_input.bulk_selection_options = (1..2).to_a.join("\n")
+      bulk_options_input.include_none_of_the_above = "true"
+      bulk_options_input.submit
+
+      expected_settings = {
+        only_one_option: "true",
```

Comment:
We should probably have a test that makes sure the value of `only_one_option` that already exists on the `answer_settings` is not overwritten.

Reply:
Done!